src/index.ts

```diff
@@ -72,6 +72,33 @@ interface BaseResult<T, E> extends Iterable<T extends Iterable<infer U> ? U : ne
 // @ts-ignore
 export declare function Err<E>(val: E): Err<E>;
 
+const isResultType = <T, E>(v: unknown): v is Result<T, E> => {
```

Comment:
This function can be replaced with using `v instanceof Err` and `v instanceof Ok`

Reply:
Perfect!